src/command_modules/azure-cli-storage/tests/test_storage_blob_scenarios.py

```diff
@@ -0,0 +1,86 @@
+
+    def verify_blob_upload_and_download(self, group, account, file_size_kb, blob_type, skip_download=False):
+        container = self.create_random_name(prefix='cont', length=24)
+        local_dir = self.create_temp_dir()
+        local_file = self.create_temp_file(file_size_kb)
+        blob_name = self.create_random_name(prefix='blob', length=24)
+        account_key = self.get_account_key(group, account)
+
+        self.set_env('AZURE_STORAGE_ACCOUNT', account)
+        self.set_env('AZURE_STORAGE_KEY', account_key)
+
+        self.cmd('storage container create -n {}'.format(container))
+
+        self.cmd('storage blob exists -n {} -c {}'.format(blob_name, container),
+                 checks=JMESPathCheck('exists', False))
+
+        self.cmd('storage blob upload -c {} -f {} -n {} --type {}'
+                 .format(container, local_file, blob_name, blob_type))
+
+        self.cmd('storage blob exists -n {} -c {}'.format(blob_name, container),
+                 checks=JMESPathCheck('exists', True))
+
+        self.cmd('storage blob show -n {} -c {}'.format(blob_name, container),
+                 checks=JMESPathCheck('name', blob_name))  # TODO: more checks
+
+        expiry = (datetime.datetime.utcnow() + datetime.timedelta(hours=1)).strftime('%Y-%m-%dT%H:%MZ')
+        sas = self.cmd('storage blob generate-sas -n {} -c {} --expiry {} '
+                       '--permissions r --https-only'.format(blob_name, container, expiry)).output
+        assert dict(pair.split('=') for pair in sas.split('&'))   # TODO: more checks
+
+        self.cmd('storage blob update -n {} -c {} --content-type application/test-content'.format(blob_name, container))
+        self.cmd('storage blob show -n {} -c {}'.format(blob_name, container),
+                 checks=JMESPathCheck('properties.contentSettings.contentType', 'application/test-content'))
+
+        self.cmd('storage blob service-properties show', checks=JMESPathCheck('hourMetrics.enabled', True))
+
+        if not skip_download:
+            downloaded = os.path.join(local_dir, 'test.file')
+            self.cmd('storage blob download -n {} -c {} --file {}'.format(blob_name, container, downloaded))
+            assert os.path.isfile(downloaded)
+
+    def get_account_key(self, group, name):
+        return self.cmd('storage account keys list -n {} -g {} --query "[0].value" -otsv'
+                        .format(name, group)).output
+
+    def create_temp_file(self, size_kb):
+        _, path = tempfile.mkstemp()
+        self.addCleanup(lambda: os.remove(path))
+
+        with open(path, mode='r+b') as file:
+            chunk = bytearray([0] * 1024)
+            for i in range(size_kb):
+                file.write(chunk)
+
+        return path
+
+    def create_temp_dir(self):
+        temp_dir = tempfile.mkdtemp()
+        self.addCleanup(lambda: shutil.rmtree(temp_dir, ignore_errors=True))
```

Comment:
Can this method be part of `azure-cli-testsdk`?
It's used a lot in several modules. And can add a flag to switch between `mkstemp` and `mkdtemp`.